src/rustfmt/main.rs

```diff
@@ -60,7 +80,8 @@ struct Opt {
     config_path: Option<PathBuf>,
     /// Rust compiler edition
     ///
-    /// Specify which edition of the compiler to use when formatting code.
+    /// Specify which edition of the compiler to use when formatting code. Uses the default
+    /// configuration's edition if not specified.
```

Comment:
I find the wording of the sentence added here a little confusing. I think the intent is to convey a default will be used but it's not clear what that will be/where the default's coming from. 

There's some outstanding questions from others whether it would make sense for a rustfmt version shipped with Rust to use a different default edition (2018) than what the corresponding rustc default is (2015), so we may just want to remove this update for now and circle back to it once those conversations have concluded.

Reply:
Can you point me to where those conversations are happening?

My intention here is to make it clear that the `--edition` flag has identical behavior as the [`edition` configuration option](https://rust-lang.github.io/rustfmt/?version=master&search=edition#edition); the existence of both ways to specify an edition could cause confusion.

Reply:
> Can you point me to where those conversations are happening?

Not really. It came up during the dev tools team's meeting this week, and I don't even have the full context myself yet. 

> My intention here is to make it clear that the --edition flag has identical behavior as the edition configuration option

Thanks for the clarification, I mentioned what I did in my previous comment because I thought it was hinting at the _value_. 

> the existence of both ways to specify an edition could cause confusion.

I'm not aware of this being an issue historically, but I don't have any objections to enhancing the help text for this option provided we can wordsmith it a bit.

It's basically just the CLI shortcut override for specifying an edition, the equivalent of `--config edition=2018`. In the absence of this flag, edition specified via the config CLI override would take precedence, and in the absence of that it would be the config file, and in the absence of that, the default. However, I don't think the help text for this flag should attempt to cover all that ground and speak to (or hint at) which edition will be used if this flag isn't specified.

I'd be okay with something in this help text that that mentions if used this is a shortcut and/or takes precedence over all other edition specifications. 

FWIW though, this isn't the only instance of there being multiple ways to set something, as we're often in need of adding a CLI opt/flag to `rustfmt` because `rustc` has an opt or flag for the same thing, or `cargo fmt` because cargo and/or other standard subcommands have the opt or flag. `rustc` obviously has and needs the `edition` flag, and so it's included directly in `rustfmt` too
```
cargo fmt --message-format json
# is the same thing as
cargo fmt -- --emit json
# or
rustfmt --emit json ....

# and
cargo fmt --check
# same as
rustfmt --check ....
# or
cargo fmt -- --check
```

Reply:
I changed the text to mention it behaves as the "edition" config option does.

In general I think it's a good idea to mention when things are duplicating work. I don't really see a disadvantage in that, as IMO the natural expectation is that options are unique (if maybe related)

Reply:
> In general I think it's a good idea to mention when things are duplicating work. I don't really see a disadvantage in that, as IMO the natural expectation is that options are unique (if maybe related)

We don't need to block this PR on it, but I'd be interested in getting a better understanding of what you're thinking (can continue inline here even after PR is merged). 

Again want to clarify that I don't have any objections to improving help text. It's just that I found the original additive wording to be confusing.

Specifically, I'm wondering:

> when things are duplicating work

Could you elaborate on what you mean by duplicated work?

> as IMO the natural expectation is that options are unique (if maybe related)

Are you saying the natural expectation would be that `--edition` does one things that's completely separate/disconnected from `edition` in the Configuration options? If so, what do you think is the natural expectation for the behaviors of those two?

Reply:
I just mean that I think it is helpful to explain the behavior and interaction of options, if there are any. As a user I could be confused whether the config option and flag behave differently; it is the only config option that also has a dedicated flag (aside the inline `config` flag), and it might be confusing until I ask someone or read the documentation. Maybe duplicating work is a poor way to put it, I just mean clarifying interactions that might be non-obvious.

Maybe a clearer example is `--emit=files` when formatting a code from stdin. In such a context, file emit means nothing. But to silently fail and write to stdout is probably also not preferable, because then the user does not understand why emit=files was not respected.

Reply:
> it is the only config option that also has a dedicated flag (aside the inline config flag)

That's true ATM, though I believe others have been considered (we almost made `--recursive` available in config files). I think the text is in a good place with this change, or at least an improvement on where it was before. Will just reiterate that edition can be specified via a cli flag (in addition to config) primarily for two reasons: parity with rustc cli and simplifying `cargo fmt`

> But to silently fail and write to stdout is probably also not preferable, because then the user does not understand why emit=files was not respected

Just want to confirm you aren't actually seeing that are you? When using an invalid emit mode with stdin mode you should see something like this:
```
Using `--emit` other than stdout is not supported with standard input.
```

Reply:
Yeah, rustfmt handles it and bails quickly, I was just using it as an example.

Reply:
whew :relieved: had me worried for a second we had a bug creep in